uefi-fs/src/lib.rs

```diff
@@ -0,0 +1,47 @@
+//! A high-level file system API for UEFI applications. It supports developers to conveniently
+//! perform the following operations:
+//! - read file to bytes
+//! - write bytes to file
+//! - create files
+//! - iterate directories and walk the file system tree
+//!
+//! Unlike UNIX-based file systems, there is no virtual file system. Thus, users perform actions on
+//! dedicated volumes: For example, the boot volume, such as a CD-rom, USB-stick, or a storage
+//! device. There are no symlinks or hard-links. Just plain paths, directories, and regular files.
+//! The way to access a volume is to open a [FileSystem].
+//!
+//! All paths are absolute and follow the FAT-like file system conventions for paths. Thus, there
+//! is no current working directory and path components like "." and ".." are not supported. In
+//! other words, the current working directory is always "/", i.e., the root, of the opened volume.
+//! This may change in the future but is currently sufficient.
+//!
+//! There are no `File` and `Path` abstractions that get publicly exported. Paths are provided as
+//! `&str` and validated internally. There are no `File` objects that are exposed to users.
+//!
+//! The difference to using the [SimpleFileSystemProtocol] directly is that the abstractions in this
+//! module take care of automatic release of resources and support developers by returning data
+//! owned on the heap (such as file info). There is no synchronization as it is untypically that
+//! users bootstrap Application Processors (AP) during the UEFI stage, i.e., before hand-off to a
+//! kernel.
+
+#![no_std]
+#![feature(error_in_core)]
+
+// Allow `dbg!` and other std-functionality in tests. This also helps IDEs to
+// provide auto-suggestions in tests.
+#[macro_use]
+#[cfg(test)]
+extern crate std;
+
+#[macro_use]
+extern crate alloc;
+
+mod dir_entry_iter;
+mod file_system;
+mod normalized_path;
+mod path;
+mod uefi_types;
+
+pub use file_system::{FileSystem, FileSystemError, FileSystemResult};
```

Comment:
I noticed from looking at the `cargo doc` output, probably want a `pub use` for `Path` here.